src/persist-client/src/batch.rs

```diff
@@ -1294,17 +1294,6 @@ pub(crate) fn validate_truncate_batch<T: Timestamp>(
         }
     }
 
-    let batch = &batch.desc;
-    if !PartialOrder::less_equal(batch.lower(), truncate.lower())
-        || PartialOrder::less_than(batch.upper(), truncate.upper())
-    {
-        return Err(InvalidUsage::InvalidBatchBounds {
-            batch_lower: batch.lower().clone(),
-            batch_upper: batch.upper().clone(),
-            append_lower: truncate.lower().clone(),
-            append_upper: truncate.upper().clone(),
-        });
-    }
```

Comment:
we can retain this check by instead verifying that the last batch in the set of batches has the same upper as the upper we are attempting to migrate to